pysmt/smtlib/parser/parser.py

```diff
@@ -1128,6 +1150,42 @@ def _cmd_assert(self, current, tokens):
         self.consume_closing(tokens, current)
         return SmtLibCommand(current, [expr])
 
+    def _cmd_assert_soft(self, current, tokens):
```

Comment:
I think this can be made more readable by not generalizing the handling of the annotations.

```python
        term_weight = None
        term_group_id = None
        curr = tokens.consume()
        while curr != ")":
            tokens.add_extra_token(curr)
            curr_parse = self.parse_atom(tokens, current)
            if curr_parse == ":weight" and term_weight is None:
                term_weight = self.get_expression(tokens)
            elif curr_parse == ":id" and term_group_id is None:
                term_group_id = self.parse_atom(tokens, "assert-soft")
            else:
                raise NotImplementedError("The current option is not currently implemented %s "
                                          "or some of them are repeated" % curr_parse)
            curr = tokens.consume()
        
        # Defaults
        if term_weight is None:
            term_weight = self.env.formula_manager.Int(1)
        if term_group_id is None:
            term_group_id =  "I"  # default identifier for soft-clause
        params = [ 
            (":weight", weight_value),
            (":id",  term_group_id),
        ]            
```